### PR TITLE
Socket click to focus

### PR DESCRIPTION
**Pull request by fakob**
* When clicking on a socket
  * the inspector opens
  * switches to the respective filter
  * scrolls to and marks the socket
* moved the toast messages to appear in the center, to not cover the drawer button
* moved the share button to the top left

https://user-images.githubusercontent.com/4619772/215353983-baccb990-f80a-4988-9960-c49baa60cf0e.mov

### CODE REVIEW

File: src/PropertyArrayContainer.tsx [hunk 1]
```diff
@@ -306,6 +309,20 @@ export const PropertyArrayContainer: React.FunctionComponent<
 
   const [configData, setConfigData] = useState(getConfigData(singleNode));
 
+  function switchFilterBasedOnSelectedSocket(node, socket) {
+    switch (true) {
```
Comment (magnificus) [inline src/PropertyArrayContainer.tsx:313]: I feel like there is a more elegant way to do this, cant you ask the socket iself what type it is? then it also does not have to be passed into socketArrayToComponent

**Review by magnificus — CHANGES_REQUESTED** [review]
Looks ait and the feature is very useful BUT left a comment on a part in the middle where I think implementation can be different to reduce dependencies and number of lines

File: src/PropertyArrayContainer.tsx (hunk 1 above)
Comment (fakob) [reply src/PropertyArrayContainer.tsx]: I am not sure that actually works. The filter has to be set outside socketArrayToComponent. However, I have now moved the checking in which array the socket is to the socketClass. Please review again.

File: src/classes/SocketClass.ts [hunk 2]
```diff
@@ -301,6 +301,17 @@ export default class Socket extends PIXI.Container {
     return this.parent as PPNode;
   }
 
+  getSocketArrayName(): string {
+    switch (true) {
```
Comment (magnificus) [inline src/classes/SocketClass.ts:305]: you already know what type the socket is, so no need to go up to the parent to check, check yourself whether you are IN OUT or TRIGGER instead

File: src/classes/NodeClass.ts [hunk 3]
```diff
@@ -619,15 +612,55 @@ export default class PPNode extends PIXI.Container {
...
+    ];
+  }
+  
+  getInputSocketByName(slotName: string): Socket {
+    return this.inputSocketArray[
+      this.inputSocketArray.findIndex((el) => el.name === slotName)
+    ];
+  }
+  
+  getInputOrTriggerSocketByName(slotName: string): Socket {
```
Comment (magnificus) [inline src/classes/NodeClass.ts:632]: might be problematic if there is a name collision between triggers and input

**Review by magnificus — APPROVED** [review]
Looking good!

File: src/classes/NodeClass.ts (hunk 3 above)
Comment (fakob) [reply src/classes/NodeClass.ts:632]: You are right. Not sure how to solve this though. The Triggers are called Trigger 1, 2 etc. Maybe let's try to not call any normal input Trigger 1?

**Pull request merged**